src/awkward/contents/unmaskedarray.py

```diff
@@ -157,12 +165,14 @@ def _getitem_nothing(self):
 
     def _getitem_at(self, where):
         if not self._backend.nplike.known_data:
+            self._touch_data(recursive=False)
```

Comment:
It was fulfilling a common pattern. If `UnmaskedArray` had any `Indexes` or other buffers directly attached to it (it doesn't), then this would be necessary to touch them before descending into its `content` (and touching whatever's there).

It's a formality.

https://github.com/scikit-hep/awkward/blob/b695e2fe78fbdb0a810a06a92e9fdd3aa3931d12/src/awkward/contents/unmaskedarray.py#L92-L94

with `recursive=False` does nothing—you're right—but it's good to treat all `Content` subclasses the same way, even if it means calling do-nothing methods from time to time.

Reply:
I think the answer to your question is "yes." It's "for posterity," in the (highly unlikely) possibility that `UnmaskedArray` might acquire an `Index` someday. If it did and we didn't treat the code in a symmetric way like this, it would be hard to find all the missing calls.